crates/datasources/src/object_store/http.rs

```diff
@@ -82,4 +90,80 @@ impl ObjStoreAccess for HttpStoreAccess {
             e_tag: None,
         })
     }
+
+    async fn into_table_provider(
```

Comment:
Http needs a special implementation as it doesn't support globbing.

so we need to create a separate provider for each location.. 

Now that we are using the file statistics, we cant just throw them into a file group like we were before
but instead actually need to do a Union
otherwise queries like `count(*)` will only use the statistics from the first location.

Reply:
we already had the `MultiSourceTableProvider` but it was in `sqlbuitlins` so i just moved it into `datasources